addons/account/models/account_account.py

```diff
@@ -575,6 +575,16 @@ def _unlink_except_account_set_on_customer(self):
                 _('You cannot remove/deactivate the account %s which is set on a customer or vendor.', account_name)
             )
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_contains_journal_items(self):
+        if self.env['account.fiscal.position.account'].search(['|', ('account_src_id', 'in', self.ids), ('account_dest_id', 'in', self.ids)], limit=1):
+            raise UserError(_('You cannot remove/deactivate the accounts "%s" which are set on the account mapping of a fiscal position.', ', '.join(f"{a.code} - {a.name}" for a in self)))
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_contains_journal_items(self):
+        if self.env['account.tax.repartition.line'].search([('account_id', 'in', self.ids)], limit=1):
+            raise UserError(_('You cannot remove/deactivate the accounts "%s" which are set on a tax repartition line.', ', '.join(f"{a.code} - {a.name}" for a in self)))
```

Comment:
I'm pretty sure this is not working because you are shadowing the first method using the exact same name with the second one